src/celeritas/em/interactor/BraggICRU73QOInteractor.hh

```diff
@@ -0,0 +1,189 @@
+ * in the low energy region according to the Bragg and ICRU73QO models. The
+ * ICRU73QO model applies to negatively charged incident particles, and the
+ * Bragg model applies to positively charged particles. The sampling of the
+ * secondary and the final state is identical in the two models, so a single
+ * interactor is used for both.
+ *
+ * \note This performs the same sampling routine as in Geant4's G4ICRU73QOModel
+ * and G4BraggModel and as documented in the Geant4 Physics Reference Manual
+ * (Release 11.1) section 12.2.1 and section 12.1.
+ */
+class BraggICRU73QOInteractor
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Mass = units::MevMass;
+    using Energy = units::MevEnergy;
+    using Momentum = units::MevMomentum;
+    //!@}
+
+  public:
+    //! Construct with shared and state data
+    inline CELER_FUNCTION
+    BraggICRU73QOInteractor(BraggICRU73QOData const& shared,
+                            ParticleTrackView const& particle,
+                            CutoffView const& cutoffs,
+                            Real3 const& inc_direction,
+                            StackAllocator<Secondary>& allocate);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Allocate space for the secondary particle
+    StackAllocator<Secondary>& allocate_;
+    // Incident direction
+    Real3 const& inc_direction_;
+    // Incident particle energy [MeV]
+    Energy inc_energy_;
+    // Incident particle momentum [MeV / c]
+    Momentum inc_momentum_;
+    // Muon mass [MeV / c^2]
+    Mass inc_mass_;
+    // Square of fractional speed of light for incident particle
+    real_type beta_sq_;
+    // Electron mass [MeV / c^2]
+    Mass electron_mass_;
+    // Secondary electron ID
+    ParticleId electron_id_;
+    // Minimum energy of the secondary electron [MeV]
+    real_type min_secondary_energy_;
+    // Maximum energy of the secondary electron [MeV]
+    real_type max_secondary_energy_;
+
+    //// HELPER FUNCTIONS ////
+
+    inline CELER_FUNCTION Energy calc_max_secondary_energy() const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ */
+CELER_FUNCTION
+BraggICRU73QOInteractor::BraggICRU73QOInteractor(
+    BraggICRU73QOData const& shared,
+    ParticleTrackView const& particle,
+    CutoffView const& cutoffs,
+    Real3 const& inc_direction,
+    StackAllocator<Secondary>& allocate)
+    : allocate_(allocate)
+    , inc_direction_(inc_direction)
+    , inc_energy_(particle.energy())
+    , inc_momentum_(particle.momentum())
+    , inc_mass_(particle.mass())
+    , beta_sq_(particle.beta_sq())
+    , electron_mass_(shared.electron_mass)
```

Comment:
Just for a consistency check, why do we use data here instead of constants::electron_mass (reverse argument for constants::proton_mass - I know our internal convention, but it is still bit confusing.

Reply:
Actually I've wondered this before too (why do we store the electron mass from the particle params in our interactor data instead of using the constant)... maybe @sethrj can clarify? The reason I used `constants::proton_mass` here is because we're not currently importing proton data.

Reply:
Well, we *could* import protons now.

Regarding the inconsistency though: CLHEP and Celeritas both define "experimentally measured" physical constants in the `constants` namespace; and those come from different sources. The cross sections and kinematics we get are from Geant4 which is using their constants, which I assume are fed into the particle data. I've tried to "consistently" use those imported values for masses and such.

The proton mass is an odd case of why this is a problem. We use the CODATA 2018 value, 1.67262192369e-24 gram = `938.272 089 43(29)` MeV. CLHEP claims to use the PDG 2019 value, which is `938.272 088 16(29)` MeV (so they're about 3 sigma off). **HOWEVER**, the value CLHEP actually uses is `938.272 013` MeV. Where does this value come from? I don't know. It could be from an older PDG evaluation that they decided not to adjust because then answers would change.

But I feel like until we generate our own cross sections, it's better to keep using whatever approximate constants Geant4 uses. Importing our "experimental constants" at runtime also allows us to compare two different code versions with the same data, or vice versa, which is an important capability for future validation studies.

Reply:
That makes sense, thanks for the explanation @sethrj! So yes, we probably should be importing the proton data. I can make a note to update this bit of code once we do. (And yes, it looks like 938.272 013(23) MeV is the 2006 CODATA value.)